src/proof/supernova.rs

```diff
@@ -228,10 +234,65 @@ where
             info!("prove_step {i}");
 
             recursive_snark
-                .prove_step(&pp.pp, step, &step.secondary_circuit())
+                .prove_step(&pp.pp, step, &secondary_circuit)
                 .unwrap();
 
             recursive_snark_option = Some(recursive_snark);
+        };
+
+        if lurk_config(None, None)
+            .perf
+            .parallelism
+            .recursive_steps
+            .is_parallel()
+        {
+            let cc = steps
+                .into_iter()
+                .map(|mf| (mf.program_counter() == 0, Mutex::new(mf)))
+                .collect::<Vec<_>>();
+
+            crossbeam::thread::scope(|s| {
+                s.spawn(|_| {
+                    // Skip the very first circuit's witness, so `prove_step` can begin immediately.
+                    // That circuit's witness will not be cached and will just be computed on-demand.
```

Comment:
Should this comment be below?

Reply:
I put this comment right before both occurrences of `skip(1)`